review_practice/react-app/src/containers/RoomList/ReservationModalBody.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useContext } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import * as Style from './style';
+
+import { SearchRoomContext } from 'pages/SearchRoom';
+
+import * as RoomAPI from 'graphql/mutation/booking';
+
+import { getNightValueFromDateString } from 'util/ConverDate';
+
+export default ({ room, searchOptions, closeModal }) => {
+    const { date, personnel } = searchOptions;
+    const { dispatchSearchOption } = useContext(SearchRoomContext);
+    const [addBooking, { data, loading, error }] = useMutation(RoomAPI.ADD_BOOKING);
+
+    useEffect(() => {
+        if (data && !data.reserveRoom.success) alert('이미 예약되어있는 숙소입니다');
+        if (data && data.reserveRoom.success) {
+            closeModal();
+            alert('예약이 완료됬습니다.');
+            dispatchSearchOption({ type: 'reset' });
+        }
+        if (loading) console.log('추가 중');
+        if (error) alert('이미 예약되어있는 숙소입니다');
+    }, [data, loading, error]);
+
+    const nights = date
+        ? getNightValueFromDateString({ startDateString: date.checkIn, endDateString: date.checkOut })
+        : 0;
+    const guest = personnel ? personnel.adult + personnel.children : 0;
+
+    const reserveButtonClick = () => {
+        const variables = { data: { roomId: room.id, date, personnel } };
+        addBooking({ variables });
+    };
+
+    return (
+        <Style.ReservationModalBody>
+            {loading && <small className="Loading-Statement">예약 처리중입니다. 잠시만 기달려주세요...</small>}
+            <h2>{room.name}</h2>
+            <hr></hr>
+            <label htmlFor="date">날짜</label>
+            {!date && <input className="Text-Center" id="date" type="text" value="날짜를 입력해주세요" readOnly />}
```

Comment:
이런 고차 컴포넌트를 하나 만든 다음에
```js
const Conditional = ({ condition, children }) => <>{condition && children}</>;
```

```suggestion
            <Conditional condition={!date}>
                <input className="Text-Center" id="date" type="text" value="날짜를 입력해주세요" readOnly />
            </Conditional>
```
이렇게 사용해주면 어떨까요?